tests/element_test_vectors.rs

```diff
@@ -385,3 +365,560 @@ fn non_equivs(file_name: &str) {
+            Format::Text(TextKind::Pretty).element_writer_for_slice(buffer)
+        }
+
+        fn make_binary_writer(buffer: &mut [u8]) -> IonResult<SliceElementWriter> {
+            Format::Binary.element_writer_for_slice(buffer)
+        }
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_text_binary(file_name: &str) {
+        good_roundtrip_text_binary(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_binary_text(file_name: &str) {
+        good_roundtrip_binary_text(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_text_pretty(file_name: &str) {
+        good_roundtrip_text_pretty(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_pretty_text(file_name: &str) {
+        good_roundtrip_pretty_text(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_pretty_binary(file_name: &str) {
+        good_roundtrip_pretty_binary(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/**/*.10n")]
+    fn ion_c_good_roundtrip_binary_pretty(file_name: &str) {
+        good_roundtrip_binary_pretty(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/bad/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/bad/**/*.10n")]
+    fn ion_c_bad(file_name: &str) {
+        bad(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/equivs/**/*.ion")]
+    #[test_resources("ion-tests/iontestdata/good/equivs/**/*.10n")]
+    fn ion_c_equivs(file_name: &str) {
+        equivs(IonCElementApi {}, file_name)
+    }
+
+    #[test_resources("ion-tests/iontestdata/good/non-equivs/**/*.ion")]
+    // no binary files exist and the macro doesn't like empty globs...
+    // see frehberg/test-generator#12
+    //#[test_resources("ion-tests/iontestdata/good/non-equivs/**/*.10n")]
+    fn ion_c_non_equivs(file_name: &str) {
+        non_equivs(IonCElementApi {}, file_name)
+    }
+}
+
+#[cfg(test)]
+mod native_element_tests {
+    use super::*;
+    use ion_rs::value::native_reader::NativeElementReader;
+    use ion_rs::value::reader::native_element_reader;
+
+    struct NativeElementApi {}
+
+    impl ElementApi for NativeElementApi {
+        type ElementReader = NativeElementReader;
+
+        fn global_skip_list() -> SkipList {
+            &[
```

Comment:
PR tour: this is the skip list for the native reader. Glancing at the failures, there seem to be a few bugs that are responsible for the lion's share. I'll start to address them in follow-on PRs.

For the moment, the `NativeElementApi` is using the native reader but the ion-c writer. Once the native reader is passing most of its `ion-tests`, I'll switch it over to a native writer. In this way I hope to minimize the research required for each bug; there should only be one untested thing being used at a time.